spec/controllers/users_controller_spec.rb

```diff
@@ -1,10 +1,13 @@
 require "rails_helper"
 
 RSpec.describe UsersController, type: :controller do
-  describe "GET #new" do
-    it "returns http success" do
-      get :new
-      expect(response).to have_http_status(:success)
+  let(:user) { create(:unconfirmed_user) }
+
+  describe "email confirmation" do
+    it "sends a confirmation email when user registers" do
+      expect {
+        UserMailer.registration_confirmation(user).deliver_now
```

Comment:
For this test I would do something like:

```ruby
expect { post :create, params } .to change { ActionMailer::Base.deliveries.size }.by 1
```

`params` is a `Hash` with the user parameters you would normally pass to the controller to create a user.

The reason for the change is that you want to test that your controller is changing the number of delivered messages. If you directly call `UserMailer` then you are not actually testing the controller. I could go to the controller's create action and remove the mailer call and the test would still pass.